Adds a really stupid syndicate bible, with a great sprite by JStheguy

Expletive opened the pull request:
:cl:
add: Syndicate Tomes have been added to traitor uplinks for 9 TC. They let an agent or an operative provide both weal and woe.
/:cl:

So I had this idea, and it's really dumb but I'm doing it anyway. After you buy it, you use it in hand and it makes you holy, like a chaplain. You can alt-click to toggle between healing and harming mode. It deals the same amount of damage as the chaplain's null rod, albeit with armor penetration. Go ahead and laugh at me. I deserve it.

Edit: Here's the sprite. I am bad at sprites.
![syndiebook1](https://cloud.githubusercontent.com/assets/9963278/26014545/dde50fc4-3711-11e7-81c6-f19306c01bb3.png)

Expletive committed:
add terrible ebook icon

Expletive committed:
Adds energy bible

Expletive committed:
Adds syndiebible to the uplink

Expletive committed:
fuck

fuck fuck fuck

Cyberboss commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +161,38 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+/obj/item/weapon/storage/book/bible/syndicate
+	icon_state ="ebook"
+	desc += "\n<span class='notice'> Alt-click to toggle between healing and harming modes.</span>"
+	var/ordained = FALSE
```
new vars after parent vars

Cyberboss commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +161,38 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+
+/obj/item/weapon/storage/book/bible/syndicate
+	icon_state ="ebook"
+	desc += "\n<span class='notice'> Alt-click to toggle between healing and harming modes.</span>"
```
you can't perform += in datum definitions

Cyberboss replied:
Do it in /Initialize()

Cyberboss commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +161,38 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	return
+
+/obj/item/storage/book/bible/syndicate/add_blood(list/blood_dna)
+	return 0
```
FALSE

Expletive committed:
Doing what cyberboss tells me to do

FYI eswords return 0 instead of FALSE for add_blood() too

Cyberboss commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -176,6 +174,10 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	var/ordained = FALSE
+
+/obj/item/weapon/storage/book/bible/syndicate/initialize()
+	desc +="\n<span class='notice'> Alt-click to toggle between healing and harming modes.</span>"
```
Capital `I` on Initialize. Also needs a . = ..()

Cyberboss commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +161,40 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	var/ordained = FALSE
+
+/obj/item/weapon/storage/book/bible/syndicate/initialize()
+	desc +="\n<span class='notice'> Alt-click to toggle between healing and harming modes.</span>"
```
Capital I on Initialize, also needs a . = ..()

InsaneHyena commented:
Should look like cult tome. The lore will be, syndicate managed to get hands on cult's weapons, but not being insane cultists, they can only use it in a limited manner.

Expletive committed:
Fixes Initialize()

I guess I don't know what the hell I'm doing.

Cyberboss commented on code/modules/uplink/uplink_item.dm:
```diff
@@ -929,6 +929,15 @@ GLOBAL_LIST_EMPTY(uplink_items) // Global list so we only initialize this once.
+			for those of little faith. Even the most steadfast athiest will be able to heal\
+			their wounded allies with it. Additionally, it's hard-light bindings can become\
+			particularly sharp, allowing the user to harm as well as heal."
+	item = /obj/weapon/storage/book/bible/syndicate
```
/obj/*item*/weapon

Expletive committed:
fixes the build

Because of course I put the wrong path in there too.

Expletive commented:
@Cyberboss Sorry that all my code is so shit that you have to keep pointing out these really dumb mistakes.

adamkad1 commented:
syndicult

ChangelingRain commented:
You should use one of the existing books as a base, they're all standardized and look nice.

Cyberboss reviewed (approved):
It's fine

JStheguy commented:
@Expletive 
![image](https://cloud.githubusercontent.com/assets/7319515/26015844/34a8486a-3728-11e7-8b85-2bdfd0d5b077.png)
Here is a sprite for this thing, if you guys want to try and slip it in before merging.

Expletive committed:
New Sprite by JStheguy

It's looks way better than the thing I tried to make. Thank you for taking pity on me.

Expletive commented:
@JStheguy That sprite is really great. Thank you!

Expletive committed:
fixes typo

Expletive committed:
New Lore

This is probably a better idea.

Expletive committed:
More Lore

changes the lore to be reverse engineered cultist tomes.

Expletive committed:
Use intent instead of alt-click to change mode

Expletive committed:
s/damn/dam/

Armhulen commented:
10 atta 10 lore

Expletive committed:
s/'fire'/BURN

fuckdamnshitcunts

KorPhaeron commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -105,7 +105,11 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	
+	if (!heal_mode)
+		..()
+		return
```
You dont need this here

Expletive replied:
Pretty sure I need this return, though. I had an issue during testing where I would attack and heal a mob at the same time, and when I added the return it fixed that.

Expletive replied:
Can now confirm, removing the return breaks the item, causing a rather long delay that's different from how a regular bible works.

Dorsisdwarf replied:
Even if you need the return, the correct syntax is 
`return ..()` ,
 not 
`..() 
return`

Expletive replied:
Oh, okay. Thank you for explaining this.

KorPhaeron commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +160,42 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+		..()
+	else
+		..(M,user,heal_mode = FALSE)
+	return
```
Or here

Expletive replied:
Can you be a bit more specific? I don't get what you're trying to explain to me.  The behavior I am looking for is that if you are on help intent you will try to heal and/or cause brain damage to the target, but if you are on any other intent, then you will attack them.

KorPhaeron replied:
You don't need those returns since you have ..() the line before it

octareenroon91 commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +160,42 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+		playsound(src.loc, 'sound/effects/snap.ogg', 50, 1)
+		H.apply_damage(5, BRUTE, pick("l_arm", "r_arm"))
+		to_chat(H, "<span class='notice'>Your name appears on the inside cover, in blood./span>")
+		var/ownername = H.name
```
`H.name` changes with masks/ID doesn't it?
Seems weird for _**something that writes itself** in your own blood_, but then again it's the Syndicate.

KorPhaeron replied:
Yeah, you want real_name

Expletive replied:
Oh. Is there a way to get a mob's true name, even if they're concealing their identity?

octareenroon91 replied:
I actually like the idea of the book taking on a forged name.

octareenroon91 replied:
`H.real_name`

Expletive committed:
Use real_name instead of name

The syndicate tome will not be fooled by trickery. It scribes it's wielders true name upon itself.

Expletive committed:
removes trailing return on attack()

Expletive committed:
fixes icon transparency

Expletive committed:
fix typo

Expletive committed:
adds spaces

ExcessiveUseOfCobblestone commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +160,41 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	damtype = BURN
+	name = "Syndicate Tome"
+	attack_verb = list("attacked", "burned", "blessed", "damned", "scorched")
+	var/ordained = FALSE
```
I think `used` would be a more fitting var. I'm trying not to bikeshed here but it will be consistent with other objects and as such can be more easily found by admins.

octareenroon91 replied:
Or even go with `var/uses = 1` so bullshit tomes can be adminspawned.

ExcessiveUseOfCobblestone commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +160,41 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	else
+		..(M,user,heal_mode = FALSE)
+
+/obj/item/storage/book/bible/syndicate/add_blood(list/blood_dna)
```
what does this do?

octareenroon91 replied:
Pretty sure it has to do with bloody weapons.

ExcessiveUseOfCobblestone replied:
Ah I see now, thanks

ExcessiveUseOfCobblestone commented:
I like this simply because you can now disguise as the chaplain and not have the obvious tell rat you out.

Expletive committed:
s/ordained/uses/

changes variable to be better for admins.

Expletive committed:
changes return

lzimann commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +159,41 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	attack_verb = list("attacked", "burned", "blessed", "damned", "scorched")
+	var/uses = 1
+
+/obj/item/weapon/storage/book/bible/syndicate/Initialize()
```
No need for this

lzimann commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +159,41 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+		to_chat(H, "<span class='notice'>Your name appears on the inside cover, in blood.</span>")
+		var/ownername = H.real_name
+		desc += "<span class='warning'>The name [ownername] is written in blood inside the cover.</span>"
+	return
```
Trailing return

lzimann commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +159,41 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+
+/obj/item/weapon/storage/book/bible/syndicate/attack(mob/living/M, mob/living/carbon/human/user, heal_mode = TRUE)
+	if (user.a_intent == INTENT_HELP)
+		..()
```
Make it return the parent

Expletive committed:
Does what lzimann says

Expletive committed:
fixthebuildfuck

lzimann commented on code/game/objects/items/weapons/storage/book.dm:
```diff
@@ -156,3 +159,38 @@ GLOBAL_LIST_INIT(bibleitemstates, list("bible", "koran", "scrapbook", "bible",
+	if (user.a_intent == INTENT_HELP)
+		return ..()
+	else
+		..(M,user,heal_mode = FALSE)
```
return this one too!

Expletive committed:
returns

I am dumb and need every tiny problem pointed out to me.

Carbonhell commented:
This pr added a storage.dmi file in the root level, was this intended..?

lzimann commented:
No, but has been removed already.
